### PR TITLE
Remove deprecation warnings

### PR DESCRIPTION
**Pull request by tsegismont**
Removes Netty deprecation warnings mostly.

Does not modify Vert.x' HttpHeaders constants (still mixing lower and upper case letters)

### CODE REVIEW

File: src/test/java/io/vertx/test/core/NetTest.java [hunk 1]
```diff
@@ -678,6 +679,7 @@ public void testCopyServerOptions() {
   }
 
   @Test
+  @SuppressWarnings("deprecation")
```
Comment (vietj) [inline src/test/java/io/vertx/test/core/NetTest.java]: why is this needed ?

**Comment by vietj** [comment]
sure, before I have a PR touching some of these classes for you ;-)

**Comment by tsegismont** [comment]
No. please 😢

Le mar. 19 juin 2018 à 18:41, Julien Viet <notifications@github.com> a
écrit :

**Comment by vietj** [comment]
what I can do is actually remove myself the deprecated stuff in the classes for my branch

**Comment by tsegismont** [comment]
The amount of code impacted is quite limited. Can't we merge this one and rebase your PR?

**Comment by vietj** [comment]
I think that given what I've rewritten, I'll wipe your code and fix again the deprecations in the code I've rewritten.

**Comment by tsegismont** [comment]
ok
